tests/integration_tests/instances.py

```diff
@@ -43,21 +52,37 @@ def emit_settings_to_log(self) -> None:
     def destroy(self):
         self.instance.delete()
 
-    def execute(self, command):
-        return self.instance.execute(command)
+    def execute(self, command, *, use_sudo=None) -> Result:
+        if use_sudo is None:
+            use_sudo = self.use_sudo
+        return self.instance.execute(command, use_sudo=use_sudo)
 
-    def pull_file(self, remote_file, local_file):
-        self.instance.pull_file(remote_file, local_file)
+    def pull_file(self, remote_path, local_path):
+        # First copy to a temporary directory because of permissions issues
+        tmp_path = _get_tmp_path()
+        self.instance.execute('cp {} {}'.format(remote_path, tmp_path))
+        self.instance.pull_file(tmp_path, local_path)
 
     def push_file(self, local_path, remote_path):
-        self.instance.push_file(local_path, remote_path)
+        # First push to a temporary directory because of permissions issues
+        tmp_path = _get_tmp_path()
+        self.instance.push_file(local_path, tmp_path)
+        self.execute('mv {} {}'.format(tmp_path, remote_path))
```

Comment:
Perhaps 'push_file' needs perms and 'owner' arguments ? using 'mv' like this will get you *some* permissions, but it seems arbitrary.  What permissions and ownership should the user expect after calling push_file ?

Reply:
I think the conclusion Dan and I came to is, if somebody needs specific permissions they can implement it.